File: api/src/main/java/com/aistore/module/checkin/entity/DriverCheckin.java
package com.aistore.module.checkin.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.FieldStrategy;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 司机每日打卡实体（表 driver_checkin）。
 * 司机/跟车员为“软引用 sys_user 或 OTHER 文本替补”二选一；车辆软引用 vehicle。
 * 可空字段标注 updateStrategy=IGNORED，使更新时可写 null（切换 用户 ↔ OTHER、清空时间等）。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("driver_checkin")
public class DriverCheckin {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long driverUserId;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private String driverOther;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long vehicleId;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long escortUserId;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private String escortOther;

    private LocalDate checkinDate;

    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private LocalDateTime clockInAt;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private LocalDateTime clockOutAt;

    /** 出勤状态枚举名 NORMAL/LATE/ABSENT/LEAVE */
    private String checkinStatus;
    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/driverCheckin.ts
/* ========================================
   司机打卡模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 出勤状态（对齐契约 CheckinStatus） */
export type CheckinStatus = 'NORMAL' | 'LATE' | 'ABSENT' | 'LEAVE';

/** 打卡详情（对齐 DriverCheckin schema） */
export interface DriverCheckin {
  id: number;
  driverUserId?: number | null;
  driverOther?: string | null;
  driverName?: string | null;
  vehicleId?: number | null;
  vehiclePlateNo?: string | null;
  escortUserId?: number | null;
  escortOther?: string | null;
  escortName?: string | null;
  checkinDate: string;
  clockInAt?: string | null;
  clockOutAt?: string | null;
  checkinStatus: CheckinStatus;
  remark?: string | null;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 打卡列表项（对齐 DriverCheckinSummary schema） */
export interface DriverCheckinSummary {
  id: number;
  driverUserId?: number | null;
  driverName?: string | null;
  vehicleId?: number | null;
  vehiclePlateNo?: string | null;
  escortName?: string | null;
  checkinDate: string;
  clockInAt?: string | null;
  clockOutAt?: string | null;
  checkinStatus: CheckinStatus;
}

/** 打卡列表分页响应 */
export interface DriverCheckinListResponse {
  items: DriverCheckinSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建打卡请求 */
export interface CreateDriverCheckinRequest {
  driverUserId?: number | null;
  driverOther?: string | null;
  vehicleId?: number | null;
  escortUserId?: number | null;
  escortOther?: string | null;
  checkinDate: string;
  clockInAt?: string | null;
  clockOutAt?: string | null;
  checkinStatus?: CheckinStatus;
  remark?: string;
}

/** 更新打卡请求 */
export interface UpdateDriverCheckinRequest {
  driverUserId?: number | null;
  driverOther?: string | null;
  vehicleId?: number | null;
  escortUserId?: number | null;
  escortOther?: string | null;
  checkinDate?: string;
  clockInAt?: string | null;
  clockOutAt?: string | null;
  checkinStatus?: CheckinStatus;
  remark?: string;
}

/** 打卡查询参数 */
export interface DriverCheckinQueryParams {
  driverUserId?: number;
  vehicleId?: number;
  checkinStatus?: CheckinStatus;
  checkinDateStart?: string;
  checkinDateEnd?: string;
  page?: number;
  pageSize?: number;
}
